### PR TITLE
Fixes morgues being able to consume ghosts, AI-eyes, etc.

### LINKED ISSUE #40589: Who ya gonna call? GHOST BUSTERS! (ghosts can be put in morgues)
Issue reported from  Round ID: 94641 (/tg/Station Sybil [ENGLISH] [US-EAST] [100% LAG FREE])  
Reporting client version: 512  
#40557: 'More orbit fixes' by ninjanomnom at commit 86399ffc06

who ya gonna call
ghostbusters

### PR DESCRIPTION
**Pull request by MrDoomBringer**
Fixes #40589

:cl: MrDoomBringer
fix: morgues have had their proton packs removed and as such no longer suck in ghosts on closing.
/:cl:

### CODE REVIEW

**Comment by SpaceManiac** [comment]
this does the opposite of what you want

**Comment by MrDoomBringer** [comment]
@SpaceManiac tell no one

**Comment by SpaceManiac** [comment]
`istype(AM, /mob) && isliving(AM)` is redundant, you just need the `isliving` check, but watch out for removing the ability to put items in

**Comment by MrDoomBringer** [comment]
@SpaceManiac if I do that, a ghost wouldn't be considered anchored and would therefore pass the !AM.anchored check.

File: code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -132,7 +132,7 @@ GLOBAL_LIST_EMPTY(bodycontainers) //Let them act as spawnpoints for revenants an
 	playsound(src, 'sound/effects/roll.ogg', 5, 1)
 	playsound(src.loc, 'sound/items/deconstruct.ogg', 50, 1)
 	for(var/atom/movable/AM in connected.loc)
-		if(!AM.anchored || AM == connected)
+		if(!AM.anchored || AM == connected && (istype(AM, /mob) && isliving(AM)))
```
Comment (vuonojenmustaturska) [inline code/game/objects/structures/morgue.dm]: isliving is istype(AM, /mob/living), you only need that one, all livings are mobs

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/morgue.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/game/objects/structures/morgue.dm]: shiet

File: code/game/objects/structures/morgue.dm [hunk 2]
```diff
@@ -132,7 +132,7 @@ GLOBAL_LIST_EMPTY(bodycontainers) //Let them act as spawnpoints for revenants an
 	playsound(src, 'sound/effects/roll.ogg', 5, 1)
 	playsound(src.loc, 'sound/items/deconstruct.ogg', 50, 1)
 	for(var/atom/movable/AM in connected.loc)
-		if(!AM.anchored || AM == connected)
+		if(!AM.anchored || AM == connected || isliving(AM))
```
Comment (ShizCalev) [inline code/game/objects/structures/morgue.dm]: did you test this

pretty sure ghosts aren't anchored, and they'd be valid still.

Comment (MrDoomBringer) [reply code/game/objects/structures/morgue.dm]: but thats why i had THE IS MOB AND ISLIVING THING AAAAAAAAAA

File: code/game/objects/structures/morgue.dm [hunk 3]
```diff
@@ -132,7 +132,7 @@ GLOBAL_LIST_EMPTY(bodycontainers) //Let them act as spawnpoints for revenants an
 	playsound(src, 'sound/effects/roll.ogg', 5, 1)
 	playsound(src.loc, 'sound/items/deconstruct.ogg', 50, 1)
 	for(var/atom/movable/AM in connected.loc)
-		if(!AM.anchored || AM == connected || isliving(AM))
+		if(!AM.anchored || AM == connected && (istype(AM, /mob) && !isliving(AM)))
```
Comment (ShizCalev) [inline code/game/objects/structures/morgue.dm]: connected is a tray.

so if it's not anchored (which ghosts aren't), or if it's a tray, and the tray is a mob, and the tray isn't living, move inside.

Comment (ShizCalev) [reply code/game/objects/structures/morgue.dm]:
```
if(ismob(AM) && !isliving(AM))
		continue
if(!AM.anchored || AM == connected)
	AM.forceMove(src)
```

File: code/game/objects/structures/morgue.dm (hunk 1 above)
Comment (ShizCalev) [inline code/game/objects/structures/morgue.dm]: if(!AM.anchored || AM == connected)
	if(ismob(AM) && !isliving(AM))
		continue
	AM.forceMove(src)

**Review by ShizCalev — CHANGES_REQUESTED** [review]
see previous review

File: code/game/objects/structures/morgue.dm [hunk 4]
```diff
@@ -132,7 +132,7 @@ GLOBAL_LIST_EMPTY(bodycontainers) //Let them act as spawnpoints for revenants an
 	playsound(src, 'sound/effects/roll.ogg', 5, 1)
 	playsound(src.loc, 'sound/items/deconstruct.ogg', 50, 1)
```
Comment (ShizCalev) [inline code/game/objects/structures/morgue.dm]: remove the .loc here while you're at it please.

File: code/game/objects/structures/morgue.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/game/objects/structures/morgue.dm]: i just added an ! cus same thing

**Comment by DaxDupont** [comment]
IT HUNGERS FOR MORE

**Review by ShizCalev — APPROVED** [review]

**Pull request merged**